src/adapters/TelnetRequestPipeline.ts

```diff
@@ -27,71 +28,105 @@ export class TelnetRequestPipeline {
 
-                if (foundDebuggerPrompt) {
-                    this.isAtDebuggerPrompt = true;
-                    if (this.hasRequests) {
-                        // There are requests waiting to be processed
-                        this.process();
-                    }
-                }
-            } else {
-                //if responseText produced a prompt, return the responseText
-                if (foundDebuggerPrompt) {
-                    //resolve the command's promise (if it cares)
-                    this.isAtDebuggerPrompt = true;
-                    this.currentRequest.onComplete(allResponseText);
-                    allResponseText = '';
-                    this.currentRequest = undefined;
-                    //try to run the next request
-                    this.process();
-                }
+        //ensure all debugger prompts appear completely on their own line
+        this.unhandledText = util.ensureDebugPromptOnOwnLine(this.unhandledText);
+
+        //discard all the "thread attached" messages as we find
+        this.unhandledText = util.removeThreadAttachedText(this.unhandledText);
+
+        //we are at a debugger prompt if the last text we received was "Brightscript Debugger>"
+        this.isAtDebuggerPrompt = util.endsWithDebuggerPrompt(this.unhandledText);
+        if (this.isProcessing) {
+            this.handleDataForIsProcessing();
+        } else {
+            //emit the unhandled text
+            this.emit('unhandled-console-output', this.unhandledText);
+            this.unhandledText = '';
+        }
+    }
+
+    private handleDataForIsProcessing() {
+        //get the first response
+        const match = /Brightscript Debugger>\s*/is.exec(this.unhandledText);
+        if (match) {
+            const response = this.cleanResponse(
+                this.unhandledText.substring(0, match.index)
+            );
+
+            logger.debug('Found response before the first "Brightscript Debugger>" prompt', { response, allText: this.unhandledText });
+            //remove the response from the unhandled text
+            this.unhandledText = this.unhandledText.substring(match.index + match[0].length);
+
+            //emit the remaining unhandled text
+            if (this.unhandledText?.length > 0) {
+                this.emit('unhandled-console-output', this.unhandledText);
             }
-        });
+            //clear the unhandled text
+            this.unhandledText = '';
+
+            //return the response to the current request
+            this.currentRequest.onComplete(response);
+            //run a new request (if there are any)
+
+            this.currentRequest = undefined;
+            this.process();
+        } else {
+            // no prompt found, wait for more data from the device
+        }
+    }
+
+    /**
+     * Remove garbage from the response
+     */
+    private cleanResponse(text: string) {
+        text = text
+            .trim()
```

Comment:
Probably shouldn't trim the text...let the caller handle trimming it if they care. we care about user logs not getting messed up.